syntax = "proto3";

package envoy.extensions.network.dns_resolver.apple.v3;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.network.dns_resolver.apple.v3";
option java_outer_classname = "AppleDnsResolverProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/network/dns_resolver/apple/v3;applev3";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: apple DNS resolver]
// [#extension: envoy.network.dns_resolver.apple]

// Configuration for apple DNS resolver.
message AppleDnsResolverConfig {
  // The resolver will avoid the system's heuristics to only return
  // IPv4 or IPv6 addresses that it considers to be "routable", instead
  // returning all possible IPv4 or IPv6 addresses. This setting is
  // ignored if the DNS lookup family is set to v4-only or v6-only.
  // This should remain false in the vast majority of cases, but may be
  // useful when performing custom filtering of addresses, such as with
  // Happy Eyeballs.
  bool include_unroutable_families = 1;
}
